packages/apps/plugins/plugin-status-bar/src/hooks/useForm.ts

```diff
@@ -0,0 +1,95 @@
+//
+// Copyright 2024 DXOS.org
+//
+import { type ChangeEvent, useCallback, useState } from 'react';
+
+import { type ValidationError } from '../util';
+
+interface FormOptions<T> {
+  initialValues: T;
+  validate: (values: any) => ValidationError[] | undefined;
+  onSubmit: (values: T) => void;
+}
+
+export const useForm = <T extends object>({ initialValues, validate, onSubmit }: FormOptions<T>) => {
```

Comment:
I think think is probably generally useful, I'd throw a `TODO: Factor out.` on here.

Might also be useful to have a jsdoc explaining intended usage.